tests/model/test_model.py

```diff
@@ -1221,6 +1221,51 @@ def test_update_star_status(self, mocker, model, event_op,
             assert (model.index['messages'][unchanged_id]['flags'] ==
                     flags_before)
 
+    @pytest.mark.parametrize('event_message_ids, indexed_ids', [
+        ([1], [1]),
+        ([1, 2], [1]),
+        ([1, 2], [1, 2]),
+        ([1], [1, 2]),
+        ([], [1, 2]),
+        ([1, 2], []),
+    ])
+    @pytest.mark.parametrize('event_op, flags_before, flags_after', [
+        ('add', [], ['read']),
+        ('add', ['read'], ['read']),
+        ('add', ['starred'], ['starred', 'read']),
+        ('add', ['read', 'starred'], ['read', 'starred']),
```

Comment:
You've got similar parametrization here as for the starred case. I understand that the `'remove` event shouldn't occur, but perhaps we should add defensive code to ensure this, and test this?

The overlap with the star code/test does make me wonder if that test and others related to `update_message_flag_status` might need reviewing. Have you checked those?

Reply:
@neiljp I've fixed a missing assertion in the other test. Apart from that I think it looks fine